Fix typing in `config_loader`

## Bug report

### Environment
- Operating System: macOS Big Sur
- Python version: 3.7

### Description of the bug 
Mypy static checking was integrated in #456. The `scanapi.config_loader` already has type annotations, but they don't pass the mypy checks.

From what I can see these errors actually point to ways this module could be improved. For example
```
scanapi/config_loader.py:33: error: Argument 1 to "construct_scalar" of "SafeConstructor"
has incompatible type "Node"; expected "ScalarNode"  [arg-type]
```
tells us that we should check that the argument of `!include` is a scalar. Currently the following include
```yaml
endpoints:
  - name: scanapi-demo
    path: ${BASE_URL}
    requests: !include {a: "b"}
```
will crash the app with
```
yaml.constructor.ConstructorError: expected a scalar node, but found mapping
```
This can be fixed by checking `if not isinstance(node, yaml.ScalarNode): ...` in `config_loader.construct_include`.

### Expected behavior ?
Mypy should pass on `scanapi.config_parser`.

### How to reproduce the the bug ?
Run the following command (after #456 has been merged or on the corresponding brach)
```
make mypy
```

--- a/scanapi/config_loader.py
+++ b/scanapi/config_loader.py
@@ -6,7 +6,7 @@ from typing import IO, Any
 
 import yaml
 
-from scanapi.errors import EmptyConfigFileError
+from scanapi.errors import BadConfigIncludeError, EmptyConfigFileError
 
 logger = logging.getLogger(__name__)
 
@@ -26,10 +26,15 @@ class Loader(yaml.SafeLoader):
 
 def construct_include(loader: Loader, node: yaml.Node) -> Any:
     """Include file referenced at node."""
-    relative_path = os.path.join(loader.root, loader.construct_scalar(node))
+    if not isinstance(node, yaml.ScalarNode):
+        include_node_str = yaml.serialize(node).strip()
+        message = f"Include tag value is not a scalar: {include_node_str}"
+        raise BadConfigIncludeError(message)
+    include_file_path = str(loader.construct_scalar(node))
+    relative_path = os.path.join(loader.root, include_file_path)
     full_path = os.path.abspath(relative_path)
 
-    with open(full_path, "r") as f:
+    with open(full_path) as f:
         return yaml.load(f, Loader)
 
 
@@ -48,4 +53,7 @@ def load_config_file(file_path):
         return data
 
 
-yaml.add_constructor("!include", construct_include, Loader)
+# Ignore parameter types due to wrong annotations in `typeshed`. Fix submitted
+# in https://github.com/python/typeshed/pull/5828
+# Can remove "type: ignore" once `types-PyYAML` has been updated.
+yaml.add_constructor("!include", construct_include, Loader)  # type: ignore

--- a/scanapi/errors.py
+++ b/scanapi/errors.py
@@ -1,5 +1,5 @@
 class MalformedSpecError(Exception):
-    pass
+    """Base class for API spec errors."""
 
 
 class HTTPMethodNotAllowedError(MalformedSpecError):
@@ -55,3 +55,7 @@ class EmptyConfigFileError(Exception):
     def __init__(self, file_path, *args):
         message = f"File '{file_path}' is empty."
         super(EmptyConfigFileError, self).__init__(message, *args)
+
+
+class BadConfigIncludeError(Exception):
+    """Raised when the value of the !include yaml tag is not a scalar."""

--- a/tests/data/api_non_scalar_include.yaml
+++ b/tests/data/api_non_scalar_include.yaml
@@ -1,0 +1,4 @@
+endpoints:
+  - name: scanapi-demo
+    path: ${BASE_URL}
+    requests: !include {a: "b"}

--- a/tests/unit/test_config_loader.py
+++ b/tests/unit/test_config_loader.py
@@ -1,7 +1,7 @@
 from pytest import mark, raises
 
 from scanapi.config_loader import load_config_file
-from scanapi.errors import EmptyConfigFileError
+from scanapi.errors import BadConfigIncludeError, EmptyConfigFileError
 
 
 @mark.describe("config loader")
@@ -64,3 +64,11 @@ class TestLoadConfigFile:
 
         assert "[Errno 2] No such file or directory: " in str(excinfo.value)
         assert "tests/data/invalid_path/include.yaml'" in str(excinfo.value)
+
+    @mark.context("include value is not a scalar")
+    @mark.it("should raise an exception")
+    def test_should_raise_exception_4(self):
+        with raises(BadConfigIncludeError) as excinfo:
+            load_config_file("tests/data/api_non_scalar_include.yaml")
+
+        assert "Include tag value is not a scalar" in str(excinfo.value)

